app/src/main/java/chat/rocket/android/chatroom/presentation/ChatRoomPresenter.kt

```diff
@@ -553,4 +554,33 @@ class ChatRoomPresenter @Inject constructor(private val view: ChatRoomView,
             }
         }
     }
-}
+
+    /**
+     * Save unfinished message, when user left chat room without sending a message. It also clears
+     * saved message from local repository when unfinishedMessage is blank.
+     *
+     * @param chatRoomId Chat room Id.
+     * @param unfinishedMessage The unfinished message to save.
+     */
+    fun saveUnfinishedMessage(chatRoomId: String, unfinishedMessage: String) {
+        val key = LocalRepository.UNFINISHED_MSG_KEY + chatRoomId
```

Comment:
We're going to support multiserver on the next release. So here the key should be:
`val key = "${currentServer}_${LocalRepository.UNFINISHED_MSG_KEY}$chatRoomId"`